crates/containerd-shim-wasmtime/src/instance.rs

```diff
@@ -168,15 +174,15 @@ impl Instance for Wasi {
         let m = prepare_module(engine.clone(), &spec, stdin, stdout, stderr)
             .map_err(|e| Error::Others(format!("error setting up module: {}", e)))?;
 
-        let mut store = Store::new(&engine, m.0);
+        let mut store = Store::new(&engine, m.0);    
 
-        debug!("instantiating instnace");
+        debug!("instantiating instance");
         let i = linker
             .instantiate(&mut store, &m.1)
             .map_err(|err| Error::Others(format!("error instantiating module: {}", err)))?;
 
         debug!("getting start function");
-        let f = i.get_func(&mut store, "_start").ok_or_else(|| {
+        let f = i.get_func(&mut store, &m.2).ok_or_else(|| {
```

Comment:
Without calling `_start`, are we be able to get stdio file descriptors properly wired up from WASI?

Reply:
Yes, wasmtime doesn't care what function you use as the entrypoint to the module, the wiring up of stdin/stdout/stderr works correctly.